crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -55,14 +59,44 @@ impl PgManager {
     /// Retrieve the validator APYs
     pub(crate) async fn fetch_validator_apys(
         &self,
+        epoch_id: Option<u64>,
         address: &NativeSuiAddress,
     ) -> Result<Option<f64>, Error> {
-        let governance_api = GovernanceReadApi::new(self.inner.clone());
+        let system_state = self.fetch_sui_system_state(epoch_id).await?;
+        let stake_subsidy_start_epoch = system_state.stake_subsidy_start_epoch;
```

Comment:
I think it would indeed make sense to have a cached `fetch_stake_subsidy_start_epoch(&self, epoch_id)` function that fetches the system state only once per epoch.

Reply:
Yes, or bundle it with the exchange rates. Not sure which way to go.

Reply:
Bundling with the exchanges rates sounds good. You can have `fetch_exchange_rates` take in just the epoch id, and do the fetching of system state inside the function.

Reply:
Actually it doesn't matter. I think either way is fine as long as we put fetching of system state into the cached function as well.